ListLibrary/MyList.cs

```diff
@@ -0,0 +1,519 @@
+
+        /// <summary>
+        /// Add to end.
+        /// </summary>
+        /// <param name="item">item</param>
+        public void Add(T item)
+        {
+            AddByIndex(_size, item);
+        }
+
+        /// <summary>
+        /// Add one item to the start of list.
+        /// </summary>
+        /// <param name="item">item</param>
+        public void AddFront(T item)
+        {
+            AddByIndex(0, item);
+        }
+
+        /// <summary>
+        /// Add one item  in pointed possition.
+        /// </summary>
+        /// <param name="item"> item </param>
+        /// <param name="pos"> position </param>
+
+        public void AddByIndex(int pos, T item)
+        {
+            if (pos < 0 || pos > _size)
+            {
+                throw new ArgumentOutOfRangeException("Index is out of range");
+            }
+
+            if (_size == _items.Length)
+            {
+                UpdateCapacity(_size + 1);
+            }
+
+            for (int i = _size; i >= 0; i--)
+            {
+                if (i == pos)
+                {
+                    _items[i] = item;
+                    break;
+                }
+                else
+                {
+                    _items[i] = _items[i - 1];
+                }
+            }
+
+            _size++;
+        }
+
+        public void Add(MyList<T> items)
+        {
+            AddByIndex(_size, items);
+        }
+
+        public void AddFront(MyList<T> items)
+        {
+            AddByIndex(0, items);
+        }
+
+        public void AddByIndex(int pos, MyList<T> items)
+        {
+            var newSize = _size + items.Count;
+
+            if (newSize >= _items.Length)
+            {
+                UpdateCapacity(newSize);
+            }
+
+            _size += items.Count;
+
+            for (int i = _size - 1, j = items.Count - 1; j >= 0; i--)
+            {
+                if (i == pos + j)
+                {
+                    _items[i] = items[j];
+                    j--;
+                }
+                else
+                {
+                    _items[i] = _items[i - items.Count];
+                }
+            }
+        }
+
+        #endregion
+
+        #region Remove
+
+        public void RemoveOneFromEnd()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size--;
+        }
+
+        public void RemoveOneFromStart()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size--;
+
+            for (int i = 0; i < _size; i++)
+            {
+                _items[i] = _items[i + 1];
+            }
+        }
+
+        public void RemoveAt(int index)
+        {
+            if (_size == 0 || index < 0 || index > _size)
+            {
+                throw new ArgumentOutOfRangeException("Size");
```

Comment:
I think, here should be two different exceptions

Reply:
fixed